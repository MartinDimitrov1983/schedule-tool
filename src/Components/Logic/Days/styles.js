import { MAGENTA } from '../../../Constants';
const styles = {
  wrapper: {
    display: 'flex',
    gap: '10px',
    margin: '0 48px',
    minHeight: '353px',
  },
  main: (frac, width, showCopies) => {
    return {
      display: 'grid',
      position: 'relative',
      gridTemplateColumns: `repeat(${frac}, 1fr)`,
      width: `${width}%`,
      borderBottom: `${showCopies && '3px solid #e11bb6'}`,
      gap: '10px',
    };
  },

  tempCopy: {
    position: 'absolute',
    bottom: '-7%',
    left: '50%',
    transform: 'translateX(-50%)',
    color: MAGENTA,
  },
};

export default styles;
