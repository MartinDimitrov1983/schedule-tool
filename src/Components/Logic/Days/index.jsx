import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { Box, Typography } from '@mui/material';
import { Card } from '../..';
import { convertDate, createTemps } from '../../../Utils';
import { WEEK_LENGTH, COPY, TEMPLATE } from '../../../Constants';
import styles from './styles';

const Days = ({
  showCopies,
  startDateCount,
  endDateCount,
  setDaysToShow,
  daysToShow,
}) => {
  const days = useSelector((state) => state.days.days);
  const template = useSelector((state) => state.days.template);
  const templateLength = useSelector((state) => state.days.templateLength);
  const isAutoComplete = useSelector((state) => state.days.isAutoComplete);

  useEffect(() => {
    setDaysToShow(days.slice(startDateCount, endDateCount));
  }, [days, startDateCount, endDateCount, setDaysToShow]);

  return (
    <Box sx={styles.wrapper}>
      {!isAutoComplete && (
        <Box
          sx={styles.main(
            daysToShow.length,
            (daysToShow.length / WEEK_LENGTH) * 100,
            showCopies,
          )}
        >
          {daysToShow.length > 0 &&
            daysToShow.map((data, index) => {
              return (
                <Card
                  day={data.day}
                  date={convertDate(data.date)}
                  dayIndex={index + startDateCount}
                  isTemplate={true}
                  isAutocomplete={false}
                />
              );
            })}
        </Box>
      )}

      {isAutoComplete &&
        createTemps(template.slice(0, WEEK_LENGTH), templateLength).map(
          (array, arrayindex) => {
            return (
              <Box
                sx={styles.main(
                  array.length,
                  (array.length / WEEK_LENGTH) * 100,
                  showCopies,
                )}
              >
                {array.map((data, index) => {
                  return (
                    <Card
                      day={
                        days[arrayindex * array.length + index + startDateCount]
                          ?.day
                      }
                      date={convertDate(
                        days[arrayindex * array.length + index + startDateCount]
                          ?.date,
                      )}
                      dayIndex={
                        arrayindex * array.length + index + startDateCount
                      }
                      isTemplate={arrayindex === 0}
                      autoCompleteData={data}
                    />
                  );
                })}
                <Typography sx={styles.tempCopy}>
                  {arrayindex === 0 ? TEMPLATE : COPY}
                </Typography>
              </Box>
            );
          },
        )}
    </Box>
  );
};

export default Days;
