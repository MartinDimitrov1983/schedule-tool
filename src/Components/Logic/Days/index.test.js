import React from 'react';
import { render } from '@testing-library/react';
import * as Redux from 'react-redux';

import Days from './index';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

test('Snapshot Days', () => {
  jest.spyOn(Redux, 'useSelector').mockReturnValue(['09:00', '12:00']);
  const { asFragment } = render(
    <Days
      showCopies={false}
      startDateCount={2}
      endDateCount={3}
      setDaysToShow={() => {}}
      daysToShow={[]}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
