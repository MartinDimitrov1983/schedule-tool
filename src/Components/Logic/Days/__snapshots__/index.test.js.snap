// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Days 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1m16q6q"
  >
    <div
      class="MuiBox-root css-1w9yb67"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-13ttcfy-MuiTypography-root"
      >
        Template
      </p>
    </div>
  </div>
</DocumentFragment>
`;
